src/Clapi/Types/Messages.hs

```diff
@@ -27,32 +29,45 @@ splitErrIdx ei = case ei of
   GlobalError -> Nothing
   PathError p -> fmap PathError <$> Path.splitHead p
   TimePointError p tpid -> fmap (flip TimePointError tpid) <$> Path.splitHead p
+  PostTypeError (TypeName ns s) -> Just (ns, PostTypeError s)
   TypeError (TypeName ns s) -> Just (ns, TypeError s)
 
 namespaceErrIdx :: Seg -> ErrorIndex Seg -> ErrorIndex TypeName
 namespaceErrIdx ns ei = case ei of
   GlobalError -> GlobalError
   PathError p -> PathError $ ns :</ p
   TimePointError p tpid -> TimePointError (ns :</ p) tpid
+  PostTypeError s -> PostTypeError (TypeName ns s)
   TypeError s -> TypeError $ TypeName ns s
 
 data MsgError a
   = MsgError {errIndex :: ErrorIndex a, errMsgTxt :: Text} deriving (Eq, Show)
 
-data DefMessage a
-  = MsgDefine a Definition
-  | MsgUndefine a
+data DefMessage ident def
+  = MsgDefine ident def
+  | MsgUndefine ident
   deriving (Show, Eq)
 
+-- FIXME: might be nicer to break this up into sub and unsub values typed by
```

Comment:
It'll be marginally fiddlier to serialise efficiently, but I think I did do that on my branch and it was relatively painless.